### DISCUSSION TITLE
Write configuration to BMS

### DISCUSSION THREAD

**Discussion by waltertheodor** [Louisvdw/dbus-serialbattery#1081] [Ideas]
System description:

	4 x Victron smartSolar MPPT 150/100 with 30kWp solar panels
	3 x Victron Multiplus II 48/5000
	4 x battery 48V/(16 x EVE LF280K 280Ah)
		BMS-A V11.26 V11.XW JK-B2A24S20P
		BMS-B V11.26 V11.XW JK-B2A24S20P
		BMS-C V10.09 V10.XW JK-B2A24S20P
		BMS-D V10.09 V10.XW JK-B2A24S20P
		BMS-X V11.261 V11.XW JK-B2A24S20P (test system)
		
	Venus-OS-Large v3.32 on Raspberrypi 4, 4GB RAM 
		Dbus-Serial-Battery 1.2.20240401 via RS485/USB
		Battery-Aggregator v3.0.66
		
I would like to configure my BMS via Node-Red and the "Custom-Control" node. In particular, I want to reset the SOC to 100% and use the "ForceChargingOff" and "ForceDischargingOff" functions. Unfortunately, this is currently only possible via Bluetooth and the app. Is there a solution to do this via the serial interface?

Walter

**Comment by mr-manuel** [comment]
No

**Reply by waltertheodor** [reply]
That is a clear statement. But is there a way to build such a solution into Dbus-Serial-Battery? There seem to be quite a few JK-BMS users who would like to have a solution for the SOC reset problem.

**Reply by mr-manuel** [reply]
If you search in the discussions and issues you find a lot of informations. It's simply not possible because JKBMS decided that via serial connection you cannot change anything. There are many other solutions to the SOC reset problem, all well described in many issues and discussions as well as in the config.default.ini.

I will add a section to the FAQ, since this is a most common question.

## solution 1
https://github.com/Louisvdw/dbus-serialbattery/blob/59a6d00c096c3e708aaf9cd827275c6500450374/etc/dbus-serialbattery/config.default.ini#L28-L57

## solution 2
https://github.com/Louisvdw/dbus-serialbattery/blob/59a6d00c096c3e708aaf9cd827275c6500450374/etc/dbus-serialbattery/config.default.ini#L171-L192

**Comment by mr-manuel** [comment]
[How can I reset the SOC to 100%](https://louisvdw.github.io/dbus-serialbattery/faq/#how-can-i-reset-the-soc-to-100)

**Comment by waltertheodor** [comment]
Many thanks for the tips. I know them and they work very well with my installation.
But why does JKBMS say that it is not possible to write values via the serial interface?
This is not true, at least for my test system (BMS-X, see first post...).
For example, I can switch charging and discharging on and off with these command sequences:
Charge On:		4E 57 00 13 00 00 00 00 02 03 00 AB 01 00 00 00 68 00 00 01 D1	Response:	4E 57 00 13 00 00 00 00 02 00 01 AB 00 00 00 00 68 00 00 01 CE
Discharge On:	4E 57 00 13 00 00 00 00 02 03 00 AC 01 00 00 00 68 00 00 01 D2	Response:	4E 57 00 13 00 00 00 00 02 00 01 AC 00 00 00 00 68 00 00 01 CF
Charge Off:		4E 57 00 13 00 00 00 00 02 03 00 AB 00 00 00 00 68 00 00 01 D0	Response:	4E 57 00 13 00 00 00 00 02 03 00 AB 00 00 00 00 68 00 00 01 D0
Discharge Off:	4E 57 00 13 00 00 00 00 02 03 00 AC 00 00 00 00 68 00 00 01 D1	Response:	4E 57 00 13 00 00 00 00 02 00 01 AC 00 00 00 00 68 00 00 01 CF

**Reply by mr-manuel** [reply]
This is the info the developers gave me. Only with the new inverter BMS it seems to be possible.

If it works for you feel free to open a PR. Best in my repository. Evaluate also the response, since there are many BMS where this won't work for sure.

**Comment by waltertheodor** [comment]
It would be enough for me if the two functions "Charge ON/OFF" and "Discharge ON/OFF" were implemented in such a way that they could be used via the Victron node "CustomControl".
For my tests I use a JK-B2A24S20P, software: V11.261, hardware: V11.XW. The command sequences work with this. I will still test whether it works with my older JK-B2A24S20P (V10.09 V10.XW).
First of all, many thanks for your offer.

**Reply by EdKok** [reply]
@waltertheodor Did you book any progress on this?
I'm also looking for a possibility for switching (Dis)Charge, preferably with Node-Red.
